backend/generalhw.pm

```diff
@@ -45,20 +45,20 @@ sub run_cmd ($self, $cmd, @extra_args) {
 
     push @full_cmd, @extra_args;
 
-    my ($stdin, $stdout, $stderr, $ret);
-
+    bmwqemu::diag("Calling $cmd");
+    my ($p, $out);
     {
         # Do not let the SIGCHLD handler of Mojo::IOLoop::ReadWriteProcess::Session steal the exit code from IPC::Run
         local $SIG{CHLD};
-        eval { $ret = IPC::Run::run(\@full_cmd, \$stdin, \$stdout, \$stderr) };
-        die "Unable to run command '@full_cmd' (deduced from test variable $cmd): $@\n" if $@;
+        $p = IPC::Run::start(\@full_cmd, '&>', \$out);
```

Comment:
Wouldn't it still make sense to distinguish between stdout and stderr?

Reply:
Then I might interfere with ordering of output.
So I removed decided not to distinguish.

Reply:
By just using `system` you'd retain that distinction without any extra effort.

Reply:
ah - you mean in regards to what os-autoinst writes to which output FD, yes.
But I hope that everything will hit the autoinst log and also in the right order.

Reply:
I meant that before stderr and stdout was collected in two different variables and printed seperately.
By using `system` it's not clear anymore what is stdout and what stderr.
So I don't understand how it would retain that distinction.
Not that I have a strong opinion, I just would like to understand.

Reply:
I guess Marius meant that system redirects stdout to parent process stdout and child stderr to parent stderr.
Which should here both end up in the log anyway.